pylint/extensions/_check_docs_utils.py

```diff
@@ -33,6 +34,67 @@ def space_indentation(s):
     return len(s) - len(s.lstrip(' '))
 
 
+def is_property(name, node):
+    """Check if the node is a property with the given name.
+
+    :param name: The name of the property to look for.
+    :type name: str
+    :param node: The node to check.
+    :type node: astroid.Node
+
+    :rtype: bool
+    :returns: Whether the given node is a property with the given name (True)
+        or not (False).
+    """
+    return (
+        isinstance(node, astroid.FunctionDef) and
+        node.name == name and
+        any(name.endswith("property") for name in node.decoratornames())
+    )
+
+
+def get_setters_property_name(node):
+    """Get the name of the property that the given node is a setter for.
+
+    :param node: The node to get the property name for.
+    :type node: str
+
+    :rtype: str or None
+    :returns: The name of the property that the node is a setter for,
+        or None if one could not be found.
+    """
+    decorators = node.decorators.nodes if node.decorators else []
+    for decorator in decorators:
+        if isinstance(decorator, astroid.Attribute) and \
+                decorator.attrname == "setter" and \
+                isinstance(decorator.expr, astroid.Name):
+            return decorator.expr.name
+
+
+def get_setters_property(node):
+    """Get the property node for the given setter node.
+
+    :param node: The node to get the property for.
+    :type node: astroid.FunctionDef
+
+    :rtype: astroid.FunctionDef or None
+    :returns: The node relating to the property of the given setter node,
+        or None if one could not be found.
+    """
+    property_ = None
+
+    property_name = get_setters_property_name(node)
+    class_node = node_frame_class(node)
+    if property_name and class_node is not None:
```

Comment:
Why the explicit check for `none`?  Can `class_node` be a Falsey value other than `None` you'd want to accept?